parachain/src/xcm.rs

```diff
@@ -0,0 +1,238 @@
+		Superuser,
+	}
+
+	impl From<ParachainDispatchOrigin> for MultiOrigin {
+		fn from(o: ParachainDispatchOrigin) -> Self {
+			match o {
+				ParachainDispatchOrigin::Parachain => MultiOrigin::Native,
+				ParachainDispatchOrigin::Signed => MultiOrigin::SovereignAccount,
+				ParachainDispatchOrigin::Root => MultiOrigin::Superuser,
+			}
+		}
+	}
+
+	impl From<MultiOrigin> for ParachainDispatchOrigin {
+		fn from(o: MultiOrigin) -> Self {
+			match o {
+				MultiOrigin::Native => ParachainDispatchOrigin::Parachain,
+				MultiOrigin::SovereignAccount => ParachainDispatchOrigin::Signed,
+				MultiOrigin::Superuser => ParachainDispatchOrigin::Root,
+			}
+		}
+	}
+
+	#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+	pub enum MultiNetwork {
+		Wildcard,
+		Identified(Vec<u8>),
+	}
+
+	#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+	pub enum MultiLocation {
+		Null,
+		X1(Junction),
+		X2(Junction, Junction),
+		X3(Junction, Junction, Junction),
+		X4(Junction, Junction, Junction, Junction),
+	}
+
+	#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+	pub enum Junction {
+		Parent,
+		Parachain { #[codec(compact)] id: u32 },
+		OpaqueRemark(Vec<u8>),
+		AccountId32 { network: MultiNetwork, id: [u8; 32] },
+		AccountIndex64 { network: MultiNetwork, #[codec(compact)] index: u64 },
+		AccountKey20 { network: MultiNetwork, key: [u8; 20] },
+		/// An instanced Pallet on a Frame-based chain.
+		PalletInstance { id: u8 },
+		/// A nondescript index within the context location.
+		GeneralIndex { #[codec(compact)] id: u128 },
+		/// A nondescript datum acting as a key within the context location.
+		GeneralKey(Vec<u8>),
+	}
+
+	impl From<Junction> for MultiLocation {
+		fn from(x: Junction) -> Self {
+			MultiLocation::X1(x)
+		}
+	}
+
+	#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+	pub enum AssetInstance {
+		Undefined,
+		Index8(u8),
+		Index16 { #[codec(compact)] id: u16 },
+		Index32 { #[codec(compact)] id: u32 },
+		Index64 { #[codec(compact)] id: u64 },
+		Index128 { #[codec(compact)] id: u128 },
+		Array4([u8; 4]),
+		Array8([u8; 8]),
+		Array16([u8; 16]),
+		Array32([u8; 32]),
+		Blob(Vec<u8>),
+	}
+
+	#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug)]
+	pub enum MultiAsset {
+		Wild,
+		WildFungible,
+		WildNonFungible,
+		WildAbstractFungible { id: Vec<u8> },
+		WildAbstractNonFungible { class: Vec<u8> },
+		WildConcreteFungible { id: MultiLocation },
+		WildConcreteNonFungible { class: MultiLocation },
+		AbstractFungible { id: Vec<u8>, #[codec(compact)] amount: u128 },
+		AbstractNonFungible { class: Vec<u8>, instance: AssetInstance },
+		ConcreteFungible { id: MultiLocation, #[codec(compact)] amount: u128 },
+		ConcreteNonFungible { class: MultiLocation, instance: AssetInstance },
+		Each(Vec<MultiAsset>),
+	}
+
+	#[derive(Clone, Eq, PartialEq, Encode, Decode)]
+	pub enum Ai {
+		Each(Vec<Ai>),
+		DepositAsset { asset: MultiAsset, dest_: MultiLocation },
```

Comment:
```suggestion
		DepositAsset { asset: MultiAsset, dest: MultiLocation },
```

TODO: Clean these up once scale codec is fixed https://github.com/paritytech/parity-scale-codec/pull/222

Reply:
parity-scale-codec 1.3.5 has been released and should fix it.